python/ray/_private/runtime_env/packaging.py

```diff
@@ -416,6 +416,13 @@ def upload_package_if_needed(
     return True
 
 
+def get_destination_dir_from_uri(uri: str, base_directory: str) -> Path:
+    """Return the local directory to which this URI should be unpacked."""
+    pkg_file = Path(_get_local_path(base_directory, uri))
```

Comment:
Is this related to this PR?

Reply:
It's needed for `modify_context` for `working_dir`.  We need to be able to get the name of the working_dir based on the URI so that we can append `cd working_dir_name` to the worker startup command.